source/extensions/path/pattern_template_lib/pattern_template.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/router/path_match_policy.h"
+
+#include "source/extensions/path/pattern_template_lib/pattern_template_internal.h"
+#include "source/extensions/path/pattern_template_lib/proto/pattern_template_rewrite_segments.pb.h"
+
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace PatternTemplate {
+
+enum class RewriteStringKind { KVariable, KLiteral };
+
+struct RewritePatternSegment {
+  RewritePatternSegment(absl::string_view segment_value, RewriteStringKind kind)
+      : segment_value(segment_value), kind(kind) {}
+  absl::string_view segment_value;
+  RewriteStringKind kind;
+};
+
+// Returns the regex pattern that is equivalent to the given url_pattern.
+// Used in the config pipeline to translate user given url pattern to
+// the safe regex Envoy can understand. Strips away any variable captures.
+absl::StatusOr<std::string> convertURLPatternSyntaxToRegex(absl::string_view url_pattern);
+
+// Helper function that parses the pattern and breaks it down to either
+// literals or variable names. To be used by ParseRewritePattern().
+// Exposed here so that the validator for the rewrite pattern can also
+// use it.
+absl::StatusOr<std::vector<RewritePatternSegment>>
+parseRewritePatternHelper(absl::string_view pattern);
```

Comment:
Consider making this an anonymous namespace method in pattern_templace.cc. I think that's the only use outside tests (and the tests could call `isValidPathTemplateRewritePattern` instead.

Reply:
I wanted to keep is more testable by calling it directly